tools/src/test/java/org/apache/kafka/tools/consumer/group/ConsumerGroupExecutor.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.tools.consumer.group;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+public class ConsumerGroupExecutor implements AutoCloseable {
+    final int numThreads;
+    final ExecutorService executor;
+    final List<ConsumerRunnable> consumers = new ArrayList<>();
+
+    public ConsumerGroupExecutor(
+            String broker,
+            int numConsumers,
+            String groupId,
+            String groupProtocol,
+            String topic,
+            String strategy,
+            Optional<String> remoteAssignor,
+            Optional<Properties> customPropsOpt,
+            boolean syncCommit
```

Comment:
The `syncCommit` param seems to always be false. If that's the case what's the purpose of it? (is it with the intention of using this same common `ConsumerGroupExecutor` class from the `ConsumerGroupCommandTest` too?)

Reply:
Hi @lianetm 
Yes, your assumption is correct. I'm pretty new to being a Committer myself, so I aimed to complete this migration with minimal alterations.

Reply:
Makes sense, let's just make sure we have a follow-up Jira to update the `ConsumerGroupCommandTest` then, so we don't leave the duplicated `ConsumerGroupExecutor` we currently have.